### PR TITLE
Feature/jc 602 page layout

### PR DESCRIPTION
**Pull request by TebyakinaEkaterina**
Made the layout of the page

### CODE REVIEW

File: apps/vanilla/src/pages/index.css [hunk 1]
```diff
@@ -1,5 +1,257 @@
...
+  display: grid;
+  grid-template-columns: repeat(auto-fill, minmax(148px, 1fr));
+  gap: var(--space-4);
+}
+
+/* Book styles */
+.book-container {
+  display: flex;
+  flex-direction: column;
+  gap: 8px;
```
Comment (SofyaKirillova) [inline apps/vanilla/src/pages/index.css:160]: We have a variable for this

File: apps/vanilla/src/pages/index.css [hunk 2]
```diff
@@ -1,5 +1,257 @@
...
+  .form-field {
+    width: var(--form-field-width-lg);
+  }
+}
+
+@media (min-width: 1920px) {
+  body {
+    --form-field-width-xl: 320px;
+
+    padding: 0 338px;
```
Comment (SofyaKirillova) [inline apps/vanilla/src/pages/index.css:243]: Can we round the value?

File: apps/vanilla/src/pages/index.css [hunk 3]
```diff
@@ -1,5 +1,257 @@
...
+  }
+
+  .filters-container {
+    right: 338px;
```
Comment (SofyaKirillova) [inline apps/vanilla/src/pages/index.css:247]: Does this value have something to do with `padding`?

File: apps/vanilla/src/pages/index.html [hunk 4]
```diff
@@ -4,13 +4,113 @@
...
+
+		<main class="container">
+
+			<div class="filters-container">
+				<p class="subtitle">Filters</p>
+				<form class="filters-form">
+					<input type="text"
+								 name="search-field"
+								 placeholder='e.g. "Refactoring"'
+								 class="form-field search-field">
```
Comment (SofyaKirillova) [inline apps/vanilla/src/pages/index.html:23]: Incorrect indents

File: apps/vanilla/src/pages/index.html [hunk 5]
```diff
@@ -4,13 +4,113 @@
...
+						</div>
+					</div>
+				</section>
+
+			</div>
+
+		</main>
+		<footer>
+			<div class="footer-logo-container">
+				<p class="logo">Books app <span class="copyright">© 2023</span></p>
```
Comment (SofyaKirillova) [inline apps/vanilla/src/pages/index.html:109]:
```suggestion
				<p class="logo">Books app <span class="copyright">&copy; 2023</span></p>
```

File: apps/vanilla/src/pages/index.html [hunk 6]
```diff
@@ -4,13 +4,113 @@
...
+				<a class="contact-link">Github</a>
+			</div>
+			<a href="#" class="top-link">Go to top ↑</a>
```
Comment (SofyaKirillova) [inline apps/vanilla/src/pages/index.html:112]: Let's use HTML-code or add an svg

File: apps/vanilla/src/pages/index.html [hunk 7]
```diff
@@ -4,13 +4,113 @@
...
+					</div>
+				</section>
+
+				<section class="content__year">
+					<p class="subtitle year-title">1980 Year</p>
+					<div class="books-container">
+						<div class="book-container">
+							<img src="./assets/images/image 4.png"
+									 alt="Book cover"
+									 class="book-cover">
```
Comment (SofyaKirillova) [inline apps/vanilla/src/pages/index.html:68]:
```suggestion
									alt="Book cover"
									class="book-cover">
```

Incorrect indents
Please, correct them in the entire file

**Review by SofyaKirillova — CHANGES_REQUESTED** [review]

File: apps/vanilla/src/pages/index.html [hunk 8]
```diff
@@ -4,13 +4,117 @@
...
+    <header class="header">
+      <p class="header__logo logo">📚 Books app</p>
+    </header>
+
+    <main class="container">
+
+      <div class="container__filters filters">
+        <p class="filters__subtitle subtitle">Filters</p>
+        <form class="filters__form">
+          <input type="text"
```
Comment (andrey-gazzaev) [inline apps/vanilla/src/pages/index.html:20]:
```suggestion
          <input type="search"
```

File: apps/vanilla/src/pages/index.html [hunk 9]
```diff
@@ -4,13 +4,117 @@
...
+  <body>
+
+    <header class="header">
+      <p class="header__logo logo">📚 Books app</p>
+    </header>
+
+    <main class="container">
+
+      <div class="container__filters filters">
+        <p class="filters__subtitle subtitle">Filters</p>
```
Comment (andrey-gazzaev) [inline apps/vanilla/src/pages/index.html:18]: This is similar to the heading

File: apps/vanilla/src/pages/index.html [hunk 10]
```diff
@@ -4,13 +4,117 @@
...
+        <form class="filters__form">
+          <input type="text"
+                 name="search-field"
+                 placeholder='e.g. "Refactoring"'
+                 class="filters__form-field filters__search-field">
+          <select name="sort-field"
```
Comment (andrey-gazzaev) [inline apps/vanilla/src/pages/index.html:24]: Don't forget to add the label wrapper.
https://wiki.saritasa.rocks/frontend/languages/html#placeholders-as-labels-are-harmful

File: apps/vanilla/src/pages/index.html [hunk 11]
```diff
@@ -4,13 +4,117 @@
...
+                   class="filters__checkbox">
+            <span class="filters__checkmark"></span>
+          </label>
+        </form>
+      </div>
+
+      <div class="content">
+
+        <section class="content__subsection subsection">
+          <p class="subsection__title subtitle">2020 Year</p>
```
Comment (andrey-gazzaev) [inline apps/vanilla/src/pages/index.html:42]: Looks like the section doesn't have a heading

File: apps/vanilla/src/pages/index.html [hunk 12]
```diff
@@ -4,13 +4,117 @@
...
+                   alt="Book cover"
+                   class="book__cover">
+              <p class="book__title">Master the Skill of Debugging CSS</p>
+            </div>
+            <div class="subsection__item book">
+              <img src="./assets/images/image 3.png"
+                   alt="Book cover"
+                   class="book__cover">
+              <p class="book__title">Blue Period</p>
+            </div>
```
Comment (andrey-gazzaev) [inline apps/vanilla/src/pages/index.html:61]: Usually the same content, like cards, is wrapped in articles

File: apps/vanilla/src/pages/index.html [hunk 13]
```diff
@@ -4,13 +4,117 @@
...
+        </section>
+
+      </div>
+
+    </main>
+
+    <footer class="footer">
+      <div class="footer__logo-container">
+        <p class="footer__logo logo">Books app <span class="footer__copyright">&copy; 2023</span></p>
+        <a class="footer__contact-link">Github</a>
```
Comment (andrey-gazzaev) [inline apps/vanilla/src/pages/index.html:114]: Let's add a title to the link

**Review by andrey-gazzaev — APPROVED** [review]
Good job 👍

File: apps/vanilla/src/global/components.css [hunk 14]
```diff
@@ -1,50 +1,345 @@
...
+  font-weight: var(--font-weight-semibold);
+  line-height: 1.3;
+  text-transform: uppercase;
+}
+
 /* Header styles */
-header {
+.header {
   --header-height: 64px;
+  --header-border-width: 1px;
```
Comment (SofyaKirillova) [inline apps/vanilla/src/global/components.css:18]: It is unnecessary to create a separate variable for a value that is used once, if there are no questions when applying it

Also check out the other spaces

File: apps/vanilla/src/global/components.css [hunk 15]
```diff
@@ -1,50 +1,345 @@
...
-  font-weight: var(--font-weight-bold);
-  line-height: 1.2;
-}
-
 /* Footer styles */
-footer {
-  border-top: 1px var(--border) solid;
+.footer {
+  --footer-border-width: 1px;
+  --footer-top-padding: 24px;
```
Comment (SofyaKirillova) [inline apps/vanilla/src/global/components.css:31]: This variable is used in different places, it is better to put it in the `:host`

File: apps/vanilla/src/global/components.css [hunk 16]
```diff
@@ -1,50 +1,345 @@
...
+}
+
+/* Search field styles */
+.filters__search-field {
+  --search-field-right-padding: 12px;
+  --search-field-left-padding: 48px;
+  --bg-position-left: 20px;
+  --bg-position-top: 14px;
+
+  /* background-image: url("./assets/icons/search.svg"); */
```
Comment (SofyaKirillova) [inline apps/vanilla/src/global/components.css:119]: Delete the comment, please.

**Review by SofyaKirillova — APPROVED** [review]
Good job

File: apps/vanilla/src/global/components.css [hunk 17]
```diff
@@ -1,28 +1,345 @@
...
+  --subsection-gap: 16px;
+
+  display: grid;
+  grid-template-columns: repeat(auto-fill, minmax(148px, 1fr));
+  gap: var(--subsection-gap);
+}
+
+/* Book styles */
+.subsection__item {
+  --subsection-item-gap: 8px;
```
Comment (karbik-creator) [inline apps/vanilla/src/global/components.css:235]: The same. Reuse variables from `variables.css`

File: apps/vanilla/src/global/components.css [hunk 18]
```diff
@@ -1,28 +1,345 @@
...
+  font-weight: var(--font-weight-medium);
+  font-size: var(--font-size-body-small);
+  line-height: 1.5;
+  overflow-x: hidden;
+}
+
+/* Responsive */
+@media (min-width: 786px) {
+  .filters__form {
+    --filters-form-gap-md: 32px;
```
Comment (karbik-creator) [inline apps/vanilla/src/global/components.css:261]: The same. Reuse variables from variables.css

**Review by karbik-creator — APPROVED** [review]

**Review by karbik-creator — CHANGES_REQUESTED** [review]
Let's fix display of large book title
image

File: apps/vanilla/src/global/components.css [hunk 19]
```diff
@@ -1,28 +1,345 @@
...
+  align-items: center;
+}
+
+.content__subsection {
+  display: flex;
+  flex-direction: column;
+}
+
+.subsection__container {
+  --subsection-gap: 16px;
```
[the remark this answers is not in the document]
Comment (TebyakinaEkaterina) [reply apps/vanilla/src/global/components.css:226]: Initially, I used global variables in the variables file like in your example, but I was advised to use local variables with a more semantic name. Or, if the value is used only in one place, do not specify it in the variable at all. What should I do in the end? Replace all local ones with global ones?

**Comment by TebyakinaEkaterina** [comment]
> Let's fix display of large book title image

I fixed it

File: apps/vanilla/src/global/components.css (hunk 19 above)
[the remark this answers is not in the document]
Comment (karbik-creator) [reply apps/vanilla/src/global/components.css:218]: These variables will still be useful to you, so let's add them right away

File: apps/vanilla/src/global/components.css [hunk 20]
```diff
@@ -1,28 +1,339 @@
...
+  border-radius: var(--book-cover-border-radius);
+}
+
+.book__title {
+  font-weight: var(--font-weight-medium);
+  font-size: var(--font-size-body-small);
+  line-height: 1.5;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  max-height: 3rem;
```
Comment (karbik-creator) [inline apps/vanilla/src/global/components.css:249]: Please check if these styles work correctly. An ellipsis should appear with large text

**Review by karbik-creator — CHANGES_REQUESTED** [review]

**Review by karbik-creator — APPROVED** [review]
Well done!

**Pull request merged**